### PR TITLE
api and swift 3 doc updates

### PR DESCRIPTION
**Pull request by EricAppel**
This PR updates the README and all inline documentation to reflect the public API updates and uses the new Swift 3 documentation comment format.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ Elevate is a JSON parsing framework that leverages Swift to make parsing simple,
 ## Requirements
 
 - iOS 8.0+ / Mac OS X 10.10+ / tvOS 9.0+ / watchOS 2.0+
-- Xcode 8.0 beta 6+
+- Xcode 8.0
```
Comment (cnoon) [inline README.md]: I did the following in AF and Willow.

```
- iOS 9.0+ / macOS 10.11+ / tvOS 9.0+ / watchOS 2.0+
- Xcode 8.0+
- Swift 3.0+
```

If you haven't actually set the deployment targets to that yet, definitely make sure to do that.

File: Source/Elevate.swift [hunk 2]
```diff
@@ -24,17 +24,15 @@
...
-
- - throws:  A ParserError.Deserialization and ParserError.Validation error if parsing fails.
- - returns: The parsed object.
- */
+// MARK: Decodable Methods
+
+/// Decodes json data at the specified key path into an object of type `T`. `T` must implement the `Decodable` protocol.
+///
+/// - parameter from:      A Data object containing encoded json data.
+/// - parameter atKeyPath: The json key path identifying the object to be decoded. Default is `""`.
```
Comment (cnoon) [inline Source/Elevate.swift]: `keyPath`.

File: Source/Elevate.swift [hunk 3]
```diff
@@ -57,16 +55,14 @@ public func decodeObject<T: Decodable>(from data: Data, atKeyPath keyPath: Strin
...
- - parameter forKeyPath: The json key path identifying the object to be parsed. Default is `""`.
-
- - throws:  A ParserError.Deserialization and ParserError.Validation error if parsing fails.
- - returns: The parsed array of objects
- */
+/// Decodes json data at the specified key path into an array of objects of type `T`. `T` must implement the
+/// `Decodable` protocol.
+///
+/// - parameter from:      A Data object containing encoded json data.
+/// - parameter atKeyPath: The json key path identifying the object to be decoded. Default is `""`.
```
Comment (cnoon) [inline Source/Elevate.swift]: `keyPath`.

File: Source/Elevate.swift [hunk 4]
```diff
@@ -75,18 +71,16 @@ public func decodeArray<T: Decodable>(from data: Data, atKeyPath keyPath: String
...
- - parameter with:        The `Decoder` instance used to parse the data.
-
- - throws:  A ParserError.Deserialization and ParserError.Validation error if parsing fails.
- - returns: The parsed object.
- */
+/// Decodes json data at the specified key path into an object of type `T` using the passed in `Decoder` instance.
+///
+/// - parameter from:      A Data object containing encoded json data.
+/// - parameter atKeyPath: The json key path identifying the object to be decoded. Default is `""`.
+/// - parameter with:      The `Decoder` instance used to decode the data.
```
Comment (cnoon) [inline Source/Elevate.swift]: `data`, `keyPath` and `decoder`.

File: Source/Elevate.swift [hunk 5]
```diff
@@ -95,17 +89,15 @@ public func decodeObject<T>(from data: Data, atKeyPath keyPath: String = "", wit
...
-
- - throws:  A ParserError.Deserialization and ParserError.Validation error if parsing fails.
- - returns: The parsed array of objects.
- */
+/// Decodes json data at the specified key path into an array of objects of type `T` using the passed in `Decoder`
+/// instance.
+///
+/// - parameter from:      A Data object containing encoded json data.
+/// - parameter atKeyPath: The json key path identifying the object to be decoded. Default is `""`.
+/// - parameter with:      The `Decoder` instance used to decode the data.
```
Comment (cnoon) [inline Source/Elevate.swift]: `data`, `keyPath`, `decoder`.

File: Source/ParserProperty.swift [hunk 6]
```diff
@@ -24,19 +24,17 @@
...
+/// - string:     Represents a Swift `String` type.
+/// - uint:       Represents a Swift `UInt` type.
+/// - int:        Represents a Swift `Int` type.
+/// - float:      Represents a Swift `Float` type.
+/// - double:     Represents a Swift `Double` type.
+/// - bool:       Represents a Swift `Bool` type.
+/// - array:      Represents a Swift `Array` type.
+/// - dictionary: Represents a Swift `Dictionary` type.
+/// - url:        Represents a Swift `URL` type.
 public enum ParserPropertyType {
```
Comment (cnoon) [inline Source/ParserProperty.swift]: `SchemaPropertyProtocol`? Apple wants you to replace `Type` with `Protocol` now I believe.

File: Source/ParserProperty.swift [hunk 7]
```diff
@@ -49,18 +47,10 @@ public enum ParserPropertyType {
     case url
 }
 
-/**
-    Represents a parser property and all its internal characteristics.
-*/
+/// Represents a parser property and all its internal characteristics.
 public struct ParserProperty {
```
Comment (cnoon) [inline Source/ParserProperty.swift]: `SchemaProperty`?

File: Source/ParserProperty.swift [hunk 8]
```diff
@@ -80,43 +70,37 @@ public struct ParserProperty {
...
+    /// NOTE: Compound key paths may be used (e.g. `address.city`). Each property name in the key path MUST be
+    /// separated by a `.` character.
+    ///
+    /// - parameter keyPath:  Key path for property.
+    /// - parameter type:     Swift object type to be validated and extracted.
+    /// - parameter optional: Specifies if the keyPath is optional. `false` by default.
+    ///
+    /// - returns: The created parser property.
     @discardableResult
     public func addProperty(keyPath: String, type: ParserPropertyType, optional: Bool = false) -> ParserProperty {
```
Comment (cnoon) [inline Source/ParserProperty.swift]: `forKeyPath` or `keyPath`? Only ask b/c we're about to the point we're locked. 🤔

File: Source/ParserProperty.swift [hunk 9]
```diff
@@ -80,43 +70,37 @@ public struct ParserProperty {
...
-    */
+    /// Creates, adds and returns a property for the specified key path, type, optionality and decodable type.
+    ///
+    /// NOTE: Compound key paths may be used (e.g. `address.city`). Each property name in the key path MUST be
+    /// separated by a `.` character.
+    ///
+    /// - parameter keyPath:       Key path for property.
+    /// - parameter type:          Swift object type to be validated and extracted.
+    /// - parameter optional:      Specifies if the keyPath is optional. `false` by default.
+    /// - parameter decodedToType: The `Decodable` type associated to the property. `nil` by default.
```
Comment (cnoon) [inline Source/ParserProperty.swift]: `decodableType`?

File: README.md (hunk 1 above)
Comment (EricAppel) [inline README.md]: macOS! made the changes, except keeping the iOS deployment target at iOS 8.0+

File: Source/ParserProperty.swift (hunk 8 above)
Comment (EricAppel) [inline Source/ParserProperty.swift]: sticking with `keyPath` to follow initializer/factory conventions

**Comment by EricAppel** [comment]
All feedback addressed!

**Pull request merged**